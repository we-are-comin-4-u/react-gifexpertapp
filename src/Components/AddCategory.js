import React, { useState } from 'react';
import PropTypes from 'prop-types';
// import { GifExpertApp } from '../GifExpertApp'

export const AddCategory = ({setCategories}) => {

   const [inputValue,setInputValue] = useState('');

   const handleInputChanges = (e) =>{
   // console.log(e.target.value);
      setInputValue(e.target.value);
      // console.log('handleInputChangeswas called sucesfully')

   }

   const handleSubmit = (e) =>{
      e.preventDefault();
      
      // console.log('handleSubmit', inputValue)

       if(inputValue.trim().length > 2){

          setCategories(catego => [ inputValue,...catego,]);
         setInputValue('');
       }

   }

   return (
      
      <form onSubmit={handleSubmit}>
         <p>{inputValue}</p>
         <input 
         type = 'text'
         value={inputValue}
         onChange={handleInputChanges}
         />

      </form>
         
      
   )
}

AddCategory.propTypes = {
   setCategories:PropTypes.func.isRequired
}

export default AddCategory;