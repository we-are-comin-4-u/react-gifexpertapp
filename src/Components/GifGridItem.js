// import { prototype } from '@wojtekmaj/enzyme-adapter-react-17'
import React from 'react';
import PropTypes from 'prop-types';




export const GifGridItem = ({title,url}) => {

   // console.log({id,title,url});
   return (
      <div className = 'card animate__animated animate__bounce'>
         <img src = {url} alt = {title}/>
         <p>{title}</p>
      </div>
   )
}


GifGridItem.propTypes = {
   title: PropTypes.string.isRequired,
   url: PropTypes.string.isRequired,
   
   
}




// import PropTypes from 'prop-types';
