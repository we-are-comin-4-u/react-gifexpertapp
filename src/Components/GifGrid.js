import React from 'react' // { useState,useEffect }
import PropTypes from 'prop-types'

import { useFetchGifs } from '../hooks/useFetchGifs'
// import { getGifs } from '../helpers/GetGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

   //  const[images,setImages] = useState([])
     const {data:images,loading} = useFetchGifs(category);

   //   console.log(loading);
   //   console.log(data);
   // useEffect(()=>{
   //    getGifs(category).then(setImages);
   // },[category]);

   
      // getGifs();

   return (
      <>
         <h3 className ="animate__fadeIn">{category}</h3>

            { loading && <p className="animate__animated animate__flash">Loading...👩‍💻</p>  }

         
       <div className = "card-grid">
          {/* <h3>{count}</h3> */}
         
            
            
               {
                  images.map(img => ( 
                     <GifGridItem 
                     key = {img.id}//acostumbrate a usar la destructuracion
                        {...img} //esto es un operador expres hay mucho ejemplos en net
                     />  
                  ))

               }
               
      </div> 
      </>
   )
}

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}