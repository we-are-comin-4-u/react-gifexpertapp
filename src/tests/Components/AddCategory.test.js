import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import {AddCategory} from '../../Components/AddCategory'
describe('testing <AddCategory/>', ()=>{
   
   const setCategories = jest.fn();
   let wrapper = shallow(<AddCategory setCategories = {setCategories}/>);

   beforeEach(()=>{
      jest.clearAllMocks();
       wrapper = shallow(<AddCategory setCategories = {setCategories}/>);
       
      //   wrapper.find('input','form').simulate(
         //  'change',{target:{value:value}},
         //  'submit',{preventDefault(){}});

   })

   test('should to show all is already',()=>{
      
      expect(wrapper).toMatchSnapshot();
      
   })

   test('should to change text box', ()=>{

      const input = wrapper.find('input');
      const value = 'Im workint on it';
      input.simulate('change', {target:{value:value}});
      expect(wrapper.find('p').text().trim()).toBe(value )

   })

   test('shouldn,t call information with submit',()=>{

      wrapper.find('form').simulate('submit',{preventDefault(){}});
      
      expect(setCategories).not.toHaveBeenCalled();

   })

   test('should to call setCategories and clean the text box',()=>{

      const value ='You,re an asshole'
      wrapper.find('input').simulate('change',{target:{value:value}});
      wrapper.find('form').simulate('submit',{preventDefault(){}});
      expect(setCategories).toHaveBeenCalled();
      expect(setCategories).toHaveBeenCalledTimes(1);
      expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
      expect(wrapper.find('input').prop('value')).toBe('');
      //   const input = wrapper.find('input');
      //   const value = '';
      //   input.simulate('change', {target:{value:value}});
      //    expect(wrapper.find('p').text().trim()).toBe(value );
      //    wrapper.find('form').simulate('submit',{preventDefault(){}});
      //    expect(setCategories).not.toHaveBeenCalled();


   })




})