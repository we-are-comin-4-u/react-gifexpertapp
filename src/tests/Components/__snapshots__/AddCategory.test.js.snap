// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <AddCategory/> should to show all is already 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
